include/cuspv/cuda_runtime.h

```diff
@@ -26,11 +26,21 @@
  * traslate CUDA API functions to corresponding ones in HIP.
  */
 
+// Catch accidental inclusion of NVidia's cuda_runtime.h header.
+#if defined(__CUDA_RUNTIME_H__) && !defined(__CHIP_CUDA_RUNTIME_H__)
+#error                                                                         \
+    "NVidia's cuda_runtime.h appears to be included (unsupported by chipStar)"
+#endif
+
 // Using same include guard name as in CUDA SDK's cuda_runtime.h so it
 // gets excluded if it happens to be in the include search paths.
 #ifndef __CUDA_RUNTIME_H__
```

Comment:
Does this now fail if we include cuspv's cuda_runtime.h twice? Shouldn't this define moved inside the __CHIP_CUDA_RUNTIME_H__ guard?

Reply:
> Does this now fail if we include cuspv's cuda_runtime.h twice?

It's designed to not fail when included multiple times. I added a test for this just now.